[2024-02-18T06:51:35.407+0000] {taskinstance.py:1957} INFO - Dependencies all met for dep_context=non-requeueable deps ti=<TaskInstance: final_project.Stage_events manual__2024-02-18T06:51:32.798684+00:00 [queued]>
[2024-02-18T06:51:35.411+0000] {taskinstance.py:1957} INFO - Dependencies all met for dep_context=requeueable deps ti=<TaskInstance: final_project.Stage_events manual__2024-02-18T06:51:32.798684+00:00 [queued]>
[2024-02-18T06:51:35.412+0000] {taskinstance.py:2171} INFO - Starting attempt 1 of 1
[2024-02-18T06:51:35.417+0000] {taskinstance.py:2192} INFO - Executing <Task(StageToRedshiftOperator): Stage_events> on 2024-02-18 06:51:32.798684+00:00
[2024-02-18T06:51:35.421+0000] {standard_task_runner.py:60} INFO - Started process 21258 to run task
[2024-02-18T06:51:35.423+0000] {standard_task_runner.py:87} INFO - Running: ['***', 'tasks', 'run', 'final_project', 'Stage_events', 'manual__2024-02-18T06:51:32.798684+00:00', '--job-id', '27', '--raw', '--subdir', 'DAGS_FOLDER/final_project.py', '--cfg-path', '/tmp/tmp_j8g3f1t']
[2024-02-18T06:51:35.426+0000] {standard_task_runner.py:88} INFO - Job 27: Subtask Stage_events
[2024-02-18T06:51:35.457+0000] {task_command.py:423} INFO - Running <TaskInstance: final_project.Stage_events manual__2024-02-18T06:51:32.798684+00:00 [running]> on host 6b28116ad14a
[2024-02-18T06:51:35.494+0000] {taskinstance.py:2481} INFO - Exporting env vars: AIRFLOW_CTX_DAG_OWNER='udacity' AIRFLOW_CTX_DAG_ID='final_project' AIRFLOW_CTX_TASK_ID='Stage_events' AIRFLOW_CTX_EXECUTION_DATE='2024-02-18T06:51:32.798684+00:00' AIRFLOW_CTX_TRY_NUMBER='1' AIRFLOW_CTX_DAG_RUN_ID='manual__2024-02-18T06:51:32.798684+00:00'
[2024-02-18T06:51:35.500+0000] {base.py:83} INFO - Using connection ID 'aws_credentials' for task execution.
[2024-02-18T06:51:35.501+0000] {connection_wrapper.py:378} INFO - AWS Connection (conn_id='aws_credentials', conn_type='aws') credentials retrieved from login and password.
[2024-02-18T06:51:35.511+0000] {base.py:83} INFO - Using connection ID 'redshift' for task execution.
[2024-02-18T06:51:42.435+0000] {sql.py:432} INFO - Running statement: create schema if not exists staging, parameters: None
[2024-02-18T06:51:45.249+0000] {stage_redshift.py:58} INFO - Drop table staging.staging_events from destination Redshift table
[2024-02-18T06:51:45.256+0000] {base.py:83} INFO - Using connection ID 'redshift' for task execution.
[2024-02-18T06:51:51.530+0000] {sql.py:432} INFO - Running statement: Drop table if exists staging.staging_events, parameters: None
[2024-02-18T06:51:53.899+0000] {stage_redshift.py:61} INFO - Create table staging.staging_events from destination Redshift table
[2024-02-18T06:51:53.902+0000] {base.py:83} INFO - Using connection ID 'redshift' for task execution.
[2024-02-18T06:51:59.843+0000] {sql.py:432} INFO - Running statement: 
    create table if not exists staging.staging_events (
        artist text,
        auth text,
        firstname text,
        gender text,
        iteminsession int,
        lastname text,
        length double precision,
        level text,
        location text,
        method text,
        page text,
        registration bigint,
        sessionid int,
        song text,
        status int,
        ts bigint,
        useragent varchar(1000),
        userid text
    )
    , parameters: None
[2024-02-18T06:52:02.507+0000] {stage_redshift.py:64} INFO - Copying data from S3 to Redshift
[2024-02-18T06:52:02.512+0000] {warnings.py:109} WARNING - /home/***/.local/lib/python3.8/site-packages/***/utils/context.py:207: AirflowContextDeprecationWarning: Accessing 'execution_date' from the template is deprecated and will be removed in a future version. Please use 'data_interval_start' or 'logical_date' instead.
  warnings.warn(_create_deprecation_warning(key, self._deprecation_replacements[key]))

[2024-02-18T06:52:02.514+0000] {warnings.py:109} WARNING - /home/***/.local/lib/python3.8/site-packages/***/utils/context.py:207: AirflowContextDeprecationWarning: Accessing 'next_ds' from the template is deprecated and will be removed in a future version. Please use '{{ data_interval_end | ds }}' instead.
  warnings.warn(_create_deprecation_warning(key, self._deprecation_replacements[key]))

[2024-02-18T06:52:02.515+0000] {warnings.py:109} WARNING - /home/***/.local/lib/python3.8/site-packages/***/utils/context.py:207: AirflowContextDeprecationWarning: Accessing 'next_ds_nodash' from the template is deprecated and will be removed in a future version. Please use '{{ data_interval_end | ds_nodash }}' instead.
  warnings.warn(_create_deprecation_warning(key, self._deprecation_replacements[key]))

[2024-02-18T06:52:02.518+0000] {warnings.py:109} WARNING - /home/***/.local/lib/python3.8/site-packages/***/utils/context.py:207: AirflowContextDeprecationWarning: Accessing 'next_execution_date' from the template is deprecated and will be removed in a future version. Please use 'data_interval_end' instead.
  warnings.warn(_create_deprecation_warning(key, self._deprecation_replacements[key]))

[2024-02-18T06:52:02.520+0000] {warnings.py:109} WARNING - /home/***/.local/lib/python3.8/site-packages/***/utils/context.py:207: AirflowContextDeprecationWarning: Accessing 'prev_ds' from the template is deprecated and will be removed in a future version.
  warnings.warn(_create_deprecation_warning(key, self._deprecation_replacements[key]))

[2024-02-18T06:52:02.521+0000] {warnings.py:109} WARNING - /home/***/.local/lib/python3.8/site-packages/***/utils/context.py:207: AirflowContextDeprecationWarning: Accessing 'prev_ds_nodash' from the template is deprecated and will be removed in a future version.
  warnings.warn(_create_deprecation_warning(key, self._deprecation_replacements[key]))

[2024-02-18T06:52:02.523+0000] {warnings.py:109} WARNING - /home/***/.local/lib/python3.8/site-packages/***/utils/context.py:207: AirflowContextDeprecationWarning: Accessing 'prev_execution_date' from the template is deprecated and will be removed in a future version.
  warnings.warn(_create_deprecation_warning(key, self._deprecation_replacements[key]))

[2024-02-18T06:52:02.525+0000] {warnings.py:109} WARNING - /home/***/.local/lib/python3.8/site-packages/***/utils/context.py:207: AirflowContextDeprecationWarning: Accessing 'prev_execution_date_success' from the template is deprecated and will be removed in a future version. Please use 'prev_data_interval_start_success' instead.
  warnings.warn(_create_deprecation_warning(key, self._deprecation_replacements[key]))

[2024-02-18T06:52:02.528+0000] {warnings.py:109} WARNING - /home/***/.local/lib/python3.8/site-packages/***/utils/context.py:207: AirflowContextDeprecationWarning: Accessing 'tomorrow_ds' from the template is deprecated and will be removed in a future version.
  warnings.warn(_create_deprecation_warning(key, self._deprecation_replacements[key]))

[2024-02-18T06:52:02.529+0000] {warnings.py:109} WARNING - /home/***/.local/lib/python3.8/site-packages/***/utils/context.py:207: AirflowContextDeprecationWarning: Accessing 'tomorrow_ds_nodash' from the template is deprecated and will be removed in a future version.
  warnings.warn(_create_deprecation_warning(key, self._deprecation_replacements[key]))

[2024-02-18T06:52:02.530+0000] {warnings.py:109} WARNING - /home/***/.local/lib/python3.8/site-packages/***/utils/context.py:207: AirflowContextDeprecationWarning: Accessing 'yesterday_ds' from the template is deprecated and will be removed in a future version.
  warnings.warn(_create_deprecation_warning(key, self._deprecation_replacements[key]))

[2024-02-18T06:52:02.531+0000] {warnings.py:109} WARNING - /home/***/.local/lib/python3.8/site-packages/***/utils/context.py:207: AirflowContextDeprecationWarning: Accessing 'yesterday_ds_nodash' from the template is deprecated and will be removed in a future version.
  warnings.warn(_create_deprecation_warning(key, self._deprecation_replacements[key]))

[2024-02-18T06:52:02.532+0000] {taskinstance.py:2699} ERROR - Task failed with exception
Traceback (most recent call last):
  File "/home/airflow/.local/lib/python3.8/site-packages/airflow/models/taskinstance.py", line 433, in _execute_task
    result = execute_callable(context=context, **execute_callable_kwargs)
  File "/opt/airflow/plugins/custom_operators/stage_redshift.py", line 67, in execute
    formatted_sql = self.copy_sql.format(
AttributeError: 'StageToRedshiftOperator' object has no attribute 'copy_sql'
[2024-02-18T06:52:02.549+0000] {taskinstance.py:1138} INFO - Marking task as FAILED. dag_id=final_project, task_id=Stage_events, execution_date=20240218T065132, start_date=20240218T065135, end_date=20240218T065202
[2024-02-18T06:52:02.562+0000] {standard_task_runner.py:107} ERROR - Failed to execute job 27 for task Stage_events ('StageToRedshiftOperator' object has no attribute 'copy_sql'; 21258)
[2024-02-18T06:52:02.590+0000] {local_task_job_runner.py:234} INFO - Task exited with return code 1
[2024-02-18T06:52:02.602+0000] {taskinstance.py:3281} INFO - 0 downstream tasks scheduled from follow-on schedule check
